backend/src/git.rs

```diff
@@ -180,26 +177,26 @@ impl Interface {
     // it creates errors that note the destructor for other values failing because of it (tree)
     pub fn delete_doc<P: AsRef<Path> + Copy>(
         &self,
+        config: Arc<HydeConfig>,
```

Comment:
Same thing here, ideally axum handlers would get the entire config, but they'd only pass along the necessary fields to business logic, eg `config.files.docs_path`